### PR TITLE
Bari/feature/admin

### PR DESCRIPTION
**Pull request by tawhidii**

### CODE REVIEW

File: src/modules/admin/admin.controller.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
+const Admin = require("./admin.model.js");
+
+// check admin exsists or not for given email
+const checkAdminExsists = async (email) => {
+  try {
+    const adminExsists = await Admin.findOne({ where: { email: email } });
```
Comment (amanullhazoha) [inline src/modules/admin/admin.controller.js]: vai exist banan ta sonsodhon koren

Comment (tawhidii) [reply src/modules/admin/admin.controller.js]: Ops .. typo .. thanks vai .. next time I will be more careful.

File: src/modules/admin/admin.controller.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+// function for singin in admin dashboard
+const adminSignIn = async (req, res) => {
+  const data = { ...req.body };
+  const response = await checkAdminExsists(data.email);
+  if (response) {
+    try {
+      data.email == response.dataValues.email &&
+      data.password == response.dataValues.password
+        ? res.send({ message: "Welcome to deemed admin dashboard !" })
```
Comment (Emu-chy) [inline src/modules/admin/admin.controller.js]: vai 41 and 42 no line question mark ar clone sign golo keno dichen boji nai?

**Review by Emu-chy — CHANGES_REQUESTED** [review]

File: src/modules/admin/admin.controller.js (hunk 2 above)
Comment (tawhidii) [reply src/modules/admin/admin.controller.js]: Vai Please check this link https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Conditional_Operator

You will get some idea.

File: src/modules/admin/admin.controller.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+  }
+};
+
+// function for singin in admin dashboard
+const adminSignIn = async (req, res) => {
+  const data = { ...req.body };
+  const response = await checkAdminExsists(data.email);
+  if (response) {
+    try {
+      data.email == response.dataValues.email &&
```
Comment (tanvinpavel) [inline src/modules/admin/admin.controller.js]: why we check the email twice. we already check it in line 37 & 37

File: src/modules/admin/admin.controller.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+  } catch (error) {
+    console.log(error);
+  }
+};
+
+// function for register admin in system.
+const registerAdmin = async (req, res) => {
+  const data = { ...req.body };
+  const checkAdmin = await checkAdminExsists(data.email);
+  if (!checkAdmin) {
```
Comment (tanvinpavel) [inline src/modules/admin/admin.controller.js]: please try to avoid so many nested if else block.

Comment (tawhidii) [reply src/modules/admin/admin.controller.js]: Ok I will

File: src/modules/admin/admin.controller.js (hunk 3 above)
Comment (tawhidii) [reply src/modules/admin/admin.controller.js]: Vai please check , I did some update. Is it ok?

**Pull request closed without merging**